src/Uno.UWP/ApplicationModel/DataTransfer/Clipboard.Android.cs

```diff
@@ -11,10 +11,120 @@ public static partial class Clipboard
 	{
 		public static void SetContent(DataPackage content)
 		{
-			if (ContextHelper.Current.GetSystemService(Context.ClipboardService) is ClipboardManager manager)
-			{
-				manager.Text = content?.Text;
+			if(content is null)
+			{
+				throw new ArgumentNullException("cannot SetContent - null parameter");
 			}
+
+			ClipData clipData=null;
+
+			if (content.Text != null)
+			{
+				clipData = ClipData.NewPlainText("clipdata", content.Text);
+			}
+
+			if (content._uri != null)
+			{
+				clipData = ClipData.NewRawUri("clipdata", Android.Net.Uri.Parse(content._uri.ToString()));
+			}
+
+			if (content._html != null)
+			{
+				string plainText = "";
+				if(content.Text != null)
+				{
+					plainText = content.Text;
+				}
+				else
+				{
+					// simple conversion from HTML to plaintext
+					plainText = content._html;
+					for(int tagStart = plainText.IndexOf("<"); tagStart > -1; tagStart = plainText.IndexOf("<", tagStart))
+					{
+						int tagEnd = plainText.IndexOf(">", tagStart);
+						if(tagEnd <0)
+						{
+							break;	// end of loop - we have start, but without end
+						}
+
+						plainText = plainText.Remove(tagStart, tagEnd - tagStart + 1);
+					}
+				}
+				clipData = ClipData.NewHtmlText("clipdata", plainText, content._html);
+			}
+
+			if(clipData is null)
+			{
+				throw new ArgumentException("cannot SetContent - no text, html nor uri data");
+			}
+			var manager = ContextHelper.Current.GetSystemService(Context.ClipboardService) as ClipboardManager;
+			manager.PrimaryClip = clipData;
+		}
+
+		public static DataPackageView GetContent()
+		{
+			var manager = ContextHelper.Current.GetSystemService(Context.ClipboardService) as ClipboardManager;
+			if(manager is null)
+			{
+				return null;
+			}
+
+			var clipData = manager.PrimaryClip;
+			//if (!clipData.Description.HasMimeType(ClipDescription.MimetypeTextPlain))
+			//	return null;
+
+			string clipText = null;
+			Uri clipUri = null;
+			string clipHtml = null;
+			for(int iLp = 0; iLp<clipData.ItemCount; iLp++)
+			{
+				var itemText = clipData.GetItemAt(iLp).Text;
+				if(itemText != null)
+				{
+					clipText = itemText;
+				}
+				var itemUri = clipData.GetItemAt(iLp).Uri;
+				if (itemUri != null)
+				{
+					clipUri = new Uri(itemUri.ToString());

```

Comment:
if `itemUri` is a `System.Uri` you may need to use `OriginalString` instead of `ToString()`

Reply:
No, itemUri is android.content.ClipData.Item (same type as itemText or itemHtml). In this loop, I extract items that has Uri/Html/Text defined.